### PR TITLE
Topological Wheels datasets

### PR DESCRIPTION
**Pull request by martinroyer**
- [x] My dataset is in its own directory (one per directory)
- [ ] This directory also contains a proper `LICENSE` file
- [x] The dataset is generated by the author of this PR
- [ ] The dataset is taken from the web, the URL is: _Please provide the url so we can check the license_

NB : 
- my links seem broken?
- the link to our paper is inaccurate, I intend to update it once it is live!

### CODE REVIEW

**Comment by mglisse** [comment]
Can we spell the directory name with (mostly) lower case letters?
I think both numpy and panda accept compressed files, so we could compress those csv files to take less space and download faster. (they aren't *huge*, but uncompressed they double the size of this repository)
I assume pandas does the right thing with the first line of the CSV files (missing entry, 'y'). Maybe we could mention in the README for those using numpy.loadtxt that they need skiprows=1 and delimiter=','? (or whatever the right way is to read this file with numpy)
Maybe mention that the first column is the time (is there a reason not to name it 't' on the first row?), if that's what it is? Unless it is standard for time series with pandas...
`X,y=pandas.read_csv('topol_wheels_0.csv', header=0, index_col=0)`
> ValueError: too many values to unpack (expected 2)

**Comment by martinroyer** [comment]
> Can we spell the directory name with (mostly) lower case letters?

Done!

> I think both numpy and panda accept compressed files, so we could compress those csv files to take less space and download faster. (they aren't _huge_, but uncompressed they double the size of this repository)

Done!

> I assume pandas does the right thing with the first line of the CSV files (missing entry, 'y'). Maybe we could mention in the README for those using numpy.loadtxt that they need skiprows=1 and delimiter=','? (or whatever the right way is to read this file with numpy)

Done!

> Maybe mention that the first column is the time (is there a reason not to name it 't' on the first row?), if that's what it is? Unless it is standard for time series with pandas...

I believe "timestamp" is standard in pandas.

> `X,y=pandas.read_csv('topol_wheels_0.csv', header=0, index_col=0)`
> 
> > ValueError: too many values to unpack (expected 2)

Oops, my bad, fixed it in the README.

**Comment by mglisse** [comment]
> > Maybe mention that the first column is the time (is there a reason not to name it 't' on the first row?), if that's what it is? Unless it is standard for time series with pandas...
> 
> I believe "timestamp" is standard in pandas.

I wonder if naming the first column has messed up anything. `dataset` is printed as
```pre
                  0          1         2  ...        62         63  is_anomaly
timestamp                                 ...                                 
0.000      1.085919   5.762392 -1.000214  ...  4.234576   2.853848           0
```
with "timestamp" on its own line. 
Accessing the elements seems to work fine though, so maybe I am just confused by pandas' stylistic choice not to write "timestamp" on the first line.

**Comment by martinroyer** [comment]
I think it makes a special case for it being the index name, so it is not really a column.

File: timeseries/topological_wheels/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+![Example time series](./tw-data-annotated.png)
+
+To open a dataset one can use:
+```python
+# pandas
+dataset = pd.read_csv(path_to_file, header=0, index_col=0, compression='gzip')
+X, y = dataset.iloc[:, :-1], dataset.iloc[:, -1]
+
+# numpy
+dataset = np.genfromtxt(path_to_file / file, delimiter=',', skip_header=1)
```
Comment (mglisse) [inline timeseries/topological_wheels/README.md:16]: Why does the pandas version have `path_to_file` and the numpy version `path_to_file / file`?

**Review by mglisse — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -18,4 +18,10 @@ For instance, [Stanford bunny LICENSE](points/bunny/LICENSE) applies to [Stanfor
...
\ No newline at end of file
+| [SO3](points/SO3/SO3_50000.off) | The SO3 points clouds have been generated with [the ISOI software](https://mitchell-web.ornl.gov/SOI/index.php) | [OFF file](https://en.wikipedia.org/wiki/OFF_(file_format)) with 50000 points in $\mathbb{R}^9$ |
+
+## Time Series
+
+| Model Name | Source | Description                                                                                                                                                                           |
+|------------|--------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [Topological Wheels](timeseries/topological-wheels/tw-data-annotated.png) | [DataShape](https://www.inria.fr/fr/datashape) | 10 CSV files where each file has 64 timeseries over 10000 timestamps each, and a $y$ ground truth indicating anomalous regime. See [README](timeseries/topological-wheels/README.md). |
```
Comment (VincentRouvreau) [inline README.md:27]: The syntax to display an image in markdown is:
```suggestion
| ![Topological Wheels](timeseries/topological-wheels/tw-data-annotated.png) | [DataShape](https://www.inria.fr/fr/datashape) | 10 CSV files where each file has 64 timeseries over 10000 timestamps each, and a $y$ ground truth indicating anomalous regime. See [README](timeseries/topological-wheels/README.md). |
```

Comment (martinroyer) [reply README.md]: hmm I shouldn't have accepted, I think I wanted to link to the image, not display it

Comment (VincentRouvreau) [reply README.md]: Oops, sorry for the misunderstanding, I was thinking you wanted to display it...
You can roll back the modification, I let you decide

File: README.md [hunk 3]
```diff
@@ -18,4 +18,10 @@ For instance, [Stanford bunny LICENSE](points/bunny/LICENSE) applies to [Stanfor
...
\ No newline at end of file
+| [SO3](points/SO3/SO3_50000.off) | The SO3 points clouds have been generated with [the ISOI software](https://mitchell-web.ornl.gov/SOI/index.php) | [OFF file](https://en.wikipedia.org/wiki/OFF_(file_format)) with 50000 points in $\mathbb{R}^9$ |
+
+## Time Series
+
+| Model Name | Source | Description                                                                                                                                                                           |
+|------------|--------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ![Topological Wheels](timeseries/topological-wheels/tw-data-annotated.png) | [DataShape](https://www.inria.fr/fr/datashape) | 10 CSV files where each file has 64 timeseries over 10000 timestamps each, and a $y$ ground truth indicating anomalous regime. See [README](timeseries/topological-wheels/README.md). |
```
Comment (martinroyer) [inline README.md:27]:
```suggestion
| [Topological Wheels](timeseries/topological-wheels/tw-data-annotated.png) | [DataShape](https://www.inria.fr/fr/datashape) | 10 CSV files where each file has 64 timeseries over 10000 timestamps each, and a $y$ ground truth indicating anomalous regime. See [README](timeseries/topological-wheels/README.md). |
```

File: README.md (hunk 2 above)
Comment (VincentRouvreau) [inline README.md:27]: I have noticed the links were not correct (`_` and not `-` in topological_wheels directory)
```suggestion
| [Topological Wheels](timeseries/topological_wheels/tw-data-annotated.png) | [DataShape](https://www.inria.fr/fr/datashape) | 10 CSV files where each file has 64 timeseries over 10000 timestamps each, and a $y$ ground truth indicating anomalous regime. See [README](timeseries/topological_wheels/README.md). |
```

**Review by VincentRouvreau — APPROVED** [review]

**Pull request merged**